# PART 3: REGIONAL PERFORMANCE INSIGHTS

In [ ]:
import pandas as pd
import numpy as np

In [ ]:
df = pd.read_csv("/content/drive/MyDrive/Colab Notebooks/Merged_orders.csv")
df.head()

Order ID  Amount  Profit  Quantity     Category      Sub-Category  \
0  B-25601    1275   -1148         7    Furniture         Bookcases   
1  B-25601      66     -12         5     Clothing             Stole   
2  B-25601       8      -2         3     Clothing       Hankerchief   
3  B-25601      80     -56         4  Electronics  Electronic Games   
4  B-25602     168    -111         2  Electronics            Phones   

   Order Date Customer Name        State       City  
0  01-04-2018        Bharat      Gujarat  Ahmedabad  
1  01-04-2018        Bharat      Gujarat  Ahmedabad  
2  01-04-2018        Bharat      Gujarat  Ahmedabad  
3  01-04-2018        Bharat      Gujarat  Ahmedabad  
4  01-04-2018         Pearl  Maharashtra       Pune

In [ ]:
top_states = df['State'].value_counts().head(5).index
top_states

Index(['Madhya Pradesh', 'Maharashtra', 'Gujarat', 'Rajasthan', 'Delhi'], dtype='object', name='State')

In [ ]:
top_states_fil = df[df['State'].isin(top_states)]
state_result = top_states_fil.groupby('State').agg(
    Totalorder = ('Order ID','count'),
    Totalsales = ('Amount','sum'),
    Avgprofit = ('Profit','mean'),
).reset_index().sort_values(by='Totalorder',ascending=False)
state_result

State  Totalorder  Totalsales  Avgprofit
2  Madhya Pradesh         340      105140  16.326471
3     Maharashtra         290       95348  21.296552
1         Gujarat          87       21058   5.344828
0           Delhi          74       22531  40.364865
4       Rajasthan          74       21149  16.986486

# Top 5 States with higher Order
1. Madhya Pradesh
2. Maharastra
3. Gujarat
4. Delhi
5. Rajasthan

In [ ]:
region_result = df.groupby(['State','City']).agg(
    Totalsales = ('Amount','sum'),
    Totalprofit = ('Profit','sum'),
    Avgprofit = ('Profit','mean'),
    Totalorder = ('Order ID','count'),
).reset_index()

In [ ]:
low_perform = region_result[(region_result['Totalsales']<region_result['Totalsales'].median()) | (region_result['Totalprofit']< region_result['Totalprofit'].median())]
low_perform = low_perform.sort_values(by='Totalprofit')
low_perform

State        City  Totalsales  Totalprofit  Avgprofit  \
22         Tamil Nadu     Chennai        6087        -2216 -88.640000   
18             Punjab  Chandigarh       12279        -1153 -25.622222   
4             Gujarat   Ahmedabad       14230         -880 -14.193548   
19          Rajasthan      Jaipur       10076         -753 -17.113636   
0      Andhra Pradesh   Hyderabad       13256         -496 -11.809524   
1               Bihar       Patna       12943         -321  -5.177419   
8   Jammu and Kashmir     Kashmir       10829            8   0.163265   
16           Nagaland      Kohima       11903          148   3.288889   
24      Uttar Pradesh     Lucknow        5502          156   4.105263   
3                 Goa         Goa        6705          370   8.604651   
21             Sikkim     Gangtok        5276          401  16.708333   
12     Madhya Pradesh       Delhi        2488          521  74.428571   
17             Punjab    Amritsar        4507          544  36.266667   
7    Himachal Pradesh       Simla        8666          656  22.620690   
6             Haryana  Chandigarh        8863         1325  50.961538   
5             Gujarat       Surat        6828         1345  53.800000   
20          Rajasthan     Udaipur       11073         2010  67.000000   

    Totalorder  
22          25  
18          45  
4           62  
19          44  
0           42  
1           62  
8           49  
16          45  
24          38  
3           43  
21          24  
12           7  
17          15  
7           29  
6           26  
5           25  
20          30

# Regional Disparities in Sales and Profitability
1. Chennai from Tamil Nadu
2. Chandigarh from Punjab
3. Ahmedabad from Gujarat
4. Jaipur from Rajasthan
5. Hyderabad from Andhra Pradesh
6. Patna from Bihar

These city's sales face a lot of loss in the total sales order and profit where in chennai there only 25 orders where placed and it's total profit is negative(-2216) which means there is no profit margin in that region so we have to cross-check operational costs, heavy discounting, or low pricing strategies responsible for the losses. Focus on product pricing and higher-margin product lines.

In [ ]:
high_perform = low_perform.sort_values(by='Avgprofit',ascending= False)
high_perform

State        City  Totalsales  Totalprofit  Avgprofit  \
12     Madhya Pradesh       Delhi        2488          521  74.428571   
20          Rajasthan     Udaipur       11073         2010  67.000000   
5             Gujarat       Surat        6828         1345  53.800000   
6             Haryana  Chandigarh        8863         1325  50.961538   
17             Punjab    Amritsar        4507          544  36.266667   
7    Himachal Pradesh       Simla        8666          656  22.620690   
21             Sikkim     Gangtok        5276          401  16.708333   
3                 Goa         Goa        6705          370   8.604651   
24      Uttar Pradesh     Lucknow        5502          156   4.105263   
16           Nagaland      Kohima       11903          148   3.288889   
8   Jammu and Kashmir     Kashmir       10829            8   0.163265   
1               Bihar       Patna       12943         -321  -5.177419   
0      Andhra Pradesh   Hyderabad       13256         -496 -11.809524   
4             Gujarat   Ahmedabad       14230         -880 -14.193548   
19          Rajasthan      Jaipur       10076         -753 -17.113636   
18             Punjab  Chandigarh       12279        -1153 -25.622222   
22         Tamil Nadu     Chennai        6087        -2216 -88.640000   

    Totalorder  
12           7  
20          30  
5           25  
6           26  
17          15  
7           29  
21          24  
3           43  
24          38  
16          45  
8           49  
1           62  
0           42  
4           62  
19          44  
18          45  
22          25

# Regions Positive in Profitability
1. Delhi from Madhya Pradesh
2. Udaipur from Rajasthan
3. Surat from Gujarat
4. Chandigarh from Haryana

These cities have good profitability but have less than 30 orders whereas Delhi has an average profit of '74.42' but only has 7 orders so focus on promoting products in these areas to gain more profits and plan strategies to attract more customers to change underperforming areas into a leading sales figures.

Cities like Udaipur and Jaipur from Rajasthan and Surat and Ahmedabad from Gujarat have different profit stats but in the same state. We also analyzed and find what was the reason for that difference and tried to apply some strategies to improve sales in low-performing regions.